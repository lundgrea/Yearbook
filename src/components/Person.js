import React, { Component } from 'react';
import PropTypes from 'prop-types'
import './Person.css';

class Person extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    const {photo, name, quote, superlative} = this.props.person
    return (
    <article>
      <img alt="yearbook photo" className="person-photo" src={photo}/> 
      <h3> {name} </h3>
      <p> {quote} </p>
      <h4> {superlative} </h4>
    </article>
    )
}
}

export default Person;

Person.propTypes = {
  person: PropTypes.object
}