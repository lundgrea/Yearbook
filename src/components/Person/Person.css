
.card {
  display: flex;
  flex-wrap: wrap;
  display: inline-block;
  justify-content: center;
  box-sizing: border-box;
  border: 3px solid black;
  padding: 10px;
  margin: 10px;
}

.person-photo{
  width: 100px;
  height: 100px;
}