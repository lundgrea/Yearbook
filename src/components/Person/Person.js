import React from 'react';
import PropTypes from 'prop-types'
import './Person.css';

const Person = ({id, photo, name, quote, superlative, deletePerson, updatePerson}) => { 
  return (
    <article key={id} className="card">
      <img src={photo} alt="yearbook" className="person-photo"></img>
      <h3 contentEditable="true" suppressContentEditableWarning={true} onChange={(event) => updatePerson(event)}> {name} </h3>
      <p contentEditable="true" suppressContentEditableWarning={true}> {quote} </p>
      <h4 contentEditable="true" suppressContentEditableWarning={true}> {superlative} </h4>
     <button onClick={() => deletePerson(id)}>X</button>
    </article>
    )
}
export default Person;

Person.propTypes = {
  person: PropTypes.object
}