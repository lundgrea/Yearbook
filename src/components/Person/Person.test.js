import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

describe('Person', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Person
      key={1}
      id={1}
      photo='./eric.jpg'
      name="Eric"
      quote="You miss all the shots you don\'t take."
      superlative="Most Likely to Win the PGA"
      deletePerson={jest.fn()}
      updatePerson={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });


  it('should call the deletePerson prop with the Person\'s id when clicked', () => {
    // Setup
    const removePersonMock = jest.fn();
    const wrapper = shallow(
      <Person
        key={1}
        id={1}
        photo='./eric.jpg'
        name="Eric"
        quote="You miss all the shots you don\'t take."
        superlative="Most Likely to Win the PGA"
        deletePerson={removePersonMock}
        updatePerson={jest.fn()}
      />
    );
  
    // Execution
    wrapper.find('.delete-button').simulate('click');
  
    // Expectation
    expect(removePersonMock).toHaveBeenCalledWith(1);
  });
});