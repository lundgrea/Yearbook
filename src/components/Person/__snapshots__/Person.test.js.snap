// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot with all data passed in correctly 1`] = `
<article
  className="card"
  id={1}
  key="1"
>
  <img
    alt="yearbook"
    className="person-photo"
    src="./eric.jpg"
  />
  <h3
    contentEditable="true"
    onChange={[Function]}
    suppressContentEditableWarning={true}
  >
     
    Eric
     
  </h3>
  <p
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
     
    You miss all the shots you don\\'t take.
     
  </p>
  <h4
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
     
    Most Likely to Win the PGA
     
  </h4>
  <button
    className="delete-button"
    onClick={[Function]}
  >
    X
  </button>
</article>
`;
